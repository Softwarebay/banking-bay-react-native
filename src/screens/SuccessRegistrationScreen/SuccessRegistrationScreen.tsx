import { useNavigation } from '@react-navigation/native';

import { SuccessRegistrationTemplate } from 'components/templates';
import type { NavigationProp } from 'navigation';

export const SuccessRegistrationScreen = () => {
	const navigation = useNavigation<NavigationProp>();

	const onDone = () => {
		navigation.navigate('Root');
	};

	return (
		<SuccessRegistrationTemplate
			primaryButtonTitle="Done"
			onPressPrimaryButton={onDone}
		/>
	);
};
