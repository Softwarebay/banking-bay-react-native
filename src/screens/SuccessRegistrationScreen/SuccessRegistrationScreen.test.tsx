import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { SuccessRegistrationScreen } from './SuccessRegistrationScreen';

const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
	useNavigation: () => ({ navigate: mockNavigate }),
}));

describe('SuccessRegistrationScreen', () => {
	it('should match snapshot', () => {
		const tree = testRenderer.create(<SuccessRegistrationScreen />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it("should display 'Done' primary button", () => {
		render(<SuccessRegistrationScreen />);

		const primaryBtn = screen.getByTestId(
			'registration-success-primary-btn-test-id'
		);

		expect(primaryBtn).toHaveTextContent('Done');
	});

	it('should navigate to DashboardScreen after pressing on primary button', () => {
		render(<SuccessRegistrationScreen />);

		const primaryBtn = screen.getByTestId(
			'registration-success-primary-btn-test-id'
		);

		fireEvent.press(primaryBtn);

		expect(mockNavigate).toHaveBeenCalledWith('Root');
	});
});
