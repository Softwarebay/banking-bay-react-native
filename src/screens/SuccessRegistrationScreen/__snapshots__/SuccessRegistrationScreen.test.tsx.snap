// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessRegistrationScreen should match snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "paddingBottom": 30,
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": undefined,
            },
            Object {
              "aspectRatio": 1.0727699530516432,
            },
            Object {
              "width": 200,
            },
          ]
        }
        testID="registration-icon-img-test-id"
      />
      <Text
        style={
          Array [
            Object {
              "color": "#1B1D4D",
            },
            Object {
              "fontFamily": "MulishBold",
              "fontSize": 44,
              "lineHeight": 48.4,
            },
            undefined,
          ]
        }
        testID="registration-success-heading-test-id"
      >
        Account created!
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 17,
              "lineHeight": 27.2,
            },
            Object {
              "marginHorizontal": 62,
              "marginTop": 20,
              "textAlign": "center",
            },
          ]
        }
        testID="registration-success-message-test-id"
      >
        Your account had been created successfully.
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          undefined,
        ]
      }
      testID="registration-success-primary-btn-test-id"
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4288142079,
              4279654301,
            ],
            "endPoint": Array [
              1,
              1,
            ],
            "locations": undefined,
            "startPoint": Array [
              0,
              0,
            ],
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "MulishSemiBold",
                "fontSize": 17,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          Done
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
</View>
`;
